### PR TITLE
[Auto_Parallel] Update auto parallel ci and support new hybrid cases

### PR DESCRIPTION
**Pull request by haohongxiang**
### PR types

Bug fixes
### PR changes

Others
### Description

[Auto_Parallel] Update auto parallel ci and support new hybrid cases (Pcard-73145)

### CODE REVIEW

File: test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py [hunk 1]
```diff
@@ -17,24 +17,46 @@
...
 
 
+class TestSemiAutoParallelDPMPStrategy(test_base.CommunicationTestDistBase):
+    def setUp(self):
+        super().setUp(num_of_devices=4, timeout=120, nnode=1)
+        self._default_envs = {
+            "dtype": "float32",
+            "seed": "2023",
+        }
+        # this test need to be run on 4-cards environment, but our CI only supports
```
Comment (GhostScreaming) [inline test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py]: 这个注释可以删了~

File: test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py [hunk 2]
```diff
@@ -17,24 +17,46 @@
...
+
 class TestSemiAutoParallelHybridStrategy(test_base.CommunicationTestDistBase):
     def setUp(self):
-        super().setUp(num_of_devices=2, timeout=120, nnode=2)
+        super().setUp(num_of_devices=8, timeout=120, nnode=1)
         self._default_envs = {
             "dtype": "float32",
             "seed": "2023",
         }
         # this test need to be run on 4-cards environment, but our CI only supports
```
Comment (GhostScreaming) [inline test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py]: 这个注释可以删了~

File: test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py (hunk 1 above)
Comment (haohongxiang) [reply test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py]: Done

File: test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py (hunk 2 above)
Comment (haohongxiang) [reply test/auto_parallel/hybrid_strategy/test_semi_auto_parallel_hybrid_strategy.py]: Done

File: paddle/scripts/paddle_build.sh [hunk 3]
```diff
@@ -2452,6 +2454,57 @@ set -x
...
+                        eight_cards_tests="^$testcase$"
+                    else
+                        eight_cards_tests="$eight_cards_tests|^$testcase$"
+                    fi
+                fi
+                is_hybrid=''
+                matchstr=''
+                testcase=''
+        done <<< "$test_cases";
+        card_test "$eight_cards_tests" 8 1
```
Comment (XieYunshen) [inline paddle/scripts/paddle_build.sh]: 这个应该是 `card_test "$eight_cards_tests" -1 1`, 这里`-1`指的是默认使用机器中的所有卡；
https://github.com/PaddlePaddle/Paddle/blob/3bbb89c982001ba6f26598c808e0db68a653e6e2/paddle/scripts/paddle_build.sh#L1447

Comment (haohongxiang) [reply paddle/scripts/paddle_build.sh]: Done

**Review by tianshuo78520a — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]
LGTM

**Pull request merged**